drivers/gnss/gnss_parse.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2023 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "gnss_parse.h"
+#include <string.h>
+#include <errno.h>
+#include <stdlib.h>
+
+#define GNSS_PARSE_NANO_KNOTS_IN_MMS               (1943840LL)
+#define GNSS_PARSE_NANO                            (1000000000LL)
+#define GNSS_PARSE_MICRO                           (1000000LL)
+#define GNSS_PARSE_MILLI                           (1000LL)
+
+int gnss_parse_dec_to_nano(const char *str, int64_t *nano)
+{
+	/* Variables */
+	int64_t sum = 0;
+	int8_t decimal = -1;
+	int8_t pos = 0;
+	int8_t start = 0;
+	int64_t increment;
+
+	/* Verify argument */
+	if (NULL == str || NULL == nano) {
```

Comment:
Could we write these other way around so `str == NULL || nano == NULL`, we typically do not have such coding style in zephyr

Reply:
I also find them odd :-) is this so the compiler fails hard if you use `=` instead of `==`?

Reply:
This is the kind of thing the developer must know as soon as possible. Usually, during the development, the NULL argument issues must be solved. It rarely occurs again accidentally in runtime. So, in my opinion, there are better ways to go than using a defensive approach for function arguments.

I suggest using: 
```c
        __ASSERT(str != NULL, "str is required");
        __ASSERT(nano != NULL, "nano is required");
```

Reply:
I suggest you do the same for all the function arguments. It helps developers a lot during the development process. Asserts are meant for that.

Reply:
Agreed. I will align this :)

I usually use the approach you describe
https://github.com/zephyrproject-rtos/zephyr/blob/bf1410bd20fa9778c9c1ebdcc8c07a6e16968045/subsys/modem/modem_chat.c#L686-L699